include/mesh.cc

```diff
@@ -0,0 +1,48 @@
+//! Create nodal pointers and assign indices based on input coordinates
+void Mesh::create_nodes(std::vector<Eigen::Vector3d> coords) {
+  unsigned index = 1;
+  for (auto& coord : coords) {
+    nodes_.emplace(std::pair<unsigned, std::shared_ptr<pipenetwork::Node>>(
+        index, std::make_shared<pipenetwork::Node>(index, coord)));
+    index++;
+  }
+}
+
+//! Create pipe pointers and assign indices based on the nodes at its ends
+void Mesh::create_pipes(std::vector<std::pair<unsigned, unsigned>> nodeids) {
```

Comment:
Use node ids as `unsigned long long`, it would be easier to create an alias for `unsigned long long`